TestInteractiveREPL is flaky

```
--- FAIL: TestSQLREPLTestSuite (14.27s)
    --- FAIL: TestSQLREPLTestSuite/TestInteractiveREPL (6.16s)
        require.go:44: 
            	Error Trace:	sql_test.go:89
            	Error:      	"show tables;
            	            	
            	            	+--------------+
            	            	| Table        |
            	            	+--------------+
            	            	
            	            	| blobs        |
            	            	| commit_blobs |
            	            	| commit_files |
            	            	| commit_trees |
            	            	| commits      |
            	            	| files        |
            	            	| ref_commits  |
            	            	| refs         |
            	            	| remotes      |
            	            	| repositories |
            	            	| tree_entries |
            	            	+--------------+
            	            	" does not contain "+--------------+
            	            	| Table        |
            	            	+--------------+
            	            	| blobs        |
            	            	| commit_blobs |
            	            	| commit_files |
            	            	| commit_trees |
            	            	| commits      |
            	            	| files        |
            	            	| ref_commits  |
            	            	| refs         |
            	            	| remotes      |
            	            	| repositories |
            	            	| tree_entries |
            	            	+--------------+
            	            	"
            	Test:       	TestSQLREPLTestSuite/TestInteractiveREPL
```

most probably there is some problem with smart line parser.

--- a/cmdtests/common.go
+++ b/cmdtests/common.go
@@ -265,12 +265,15 @@ func (sl *StreamLinifier) Linify(in chan string) chan string {
 
 				lines := strings.Split(sl.pending+normalizeNewLine(s), "\n")
 				sl.pending = ""
-
 				for i, l := range lines {
-					if i == len(lines) && l != "" {
-						sl.pending = l
+					if i == len(lines)-1 {
+						if l != "" {
+							sl.pending = l
+						}
+
 						break
 					}
+
 					out <- l
 				}
 			}

--- a/cmdtests/common_test.go
+++ b/cmdtests/common_test.go
@@ -1,0 +1,114 @@
+// +build integration
+
+package cmdtests_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/src-d/engine/cmdtests"
+	"github.com/stretchr/testify/suite"
+)
+
+type StreamLinifierTestSuite struct {
+	suite.Suite
+}
+
+func TestStreamLinifierTestSuite(t *testing.T) {
+	suite.Run(t, &StreamLinifierTestSuite{})
+}
+
+func (s *StreamLinifierTestSuite) TestOneLineOnOneMessage() {
+	require := s.Require()
+
+	sl := cmdtests.NewStreamLinifier(500 * time.Millisecond)
+
+	in := make(chan string)
+	messagesNum := 3
+
+	go func() {
+		for i := 1; i <= messagesNum; i++ {
+			in <- fmt.Sprintf("line %d\n", i)
+		}
+
+		close(in)
+	}()
+
+	out := sl.Linify(in)
+	counter := 0
+	for m := range out {
+		counter++
+		require.Equal(m, fmt.Sprintf("line %d", counter))
+	}
+
+	require.Equal(messagesNum, counter)
+}
+
+func (s *StreamLinifierTestSuite) TestOneLineOnMultiMessages() {
+	require := s.Require()
+
+	sl := cmdtests.NewStreamLinifier(1 * time.Second)
+
+	in := make(chan string)
+
+	go func() {
+		in <- "line 1 (1/3)|"
+		in <- "line 1 (2/3)|"
+		in <- "line 1 (3/3)\n"
+
+		in <- "line 2 (1/2)|"
+		in <- "line 2 (2/2)\n"
+
+		in <- "line 3 (1/1)\n"
+
+		close(in)
+	}()
+
+	out := sl.Linify(in)
+	require.Equal(<-out, "line 1 (1/3)|line 1 (2/3)|line 1 (3/3)")
+	require.Equal(<-out, "line 2 (1/2)|line 2 (2/2)")
+	require.Equal(<-out, "line 3 (1/1)")
+}
+
+func (s *StreamLinifierTestSuite) TestMultiLinesOnOneMessage() {
+	require := s.Require()
+
+	sl := cmdtests.NewStreamLinifier(1 * time.Second)
+
+	in := make(chan string)
+
+	go func() {
+		in <- "line 1\nline 2\nline 3\n"
+
+		close(in)
+	}()
+
+	out := sl.Linify(in)
+	require.Equal(<-out, "line 1")
+	require.Equal(<-out, "line 2")
+	require.Equal(<-out, "line 3")
+}
+
+func (s *StreamLinifierTestSuite) TestSlowPendingMessage() {
+	require := s.Require()
+
+	sl := cmdtests.NewStreamLinifier(1 * time.Second)
+
+	in := make(chan string)
+
+	go func() {
+		in <- "line 1 (1/3)|"
+		time.Sleep(1 * time.Second)
+		in <- "line 1 (2/3)|"
+		time.Sleep(1 * time.Second)
+		in <- "line 1 (3/3)\n"
+
+		close(in)
+	}()
+
+	out := sl.Linify(in)
+	require.Equal(<-out, "line 1 (1/3)|")
+	require.Equal(<-out, "line 1 (2/3)|")
+	require.Equal(<-out, "line 1 (3/3)")
+}

